content/terraform/v1.12.x/docs/language/parameterize/configure.mdx

```diff
@@ -0,0 +1,428 @@
+
+## Define locals to reuse expressions
+
+> **Hands-on:** Try the [Simplify Terraform Configuration with Locals](/terraform/tutorials/configuration-language/locals?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.
+
+Local values are similar to function-scoped variables in other programming languages. Local values assign names to expressions, letting you use the name multiple times within a module instead of repeating that expression.
+
+Define a `locals` block when you want to reuse an expression throughout your configuration. You can define the `locals` block in any module, and the value you assign can be any valid Terraform expression, and can reference the following:
+
+- Variables
+- Resource attributes
+- Function outputs
+- Other local values
+
+For example, you could define a `locals` block to create a consistent naming convention, identify your primary subnet, and to set aside environmental configuration settings:
+
+<CodeBlockConfig hideClipboard>
+
+```hcl
+locals {
+  # Naming convention
+  resource_name = "${var.project_name}-${var.environment}"
+
+  # Process the subnet list
+  primary_public_subnet = var.subnet_ids[0]
+  subnet_count          = length(var.subnet_ids)
+
+  # Environmental deployment settings
+  is_production      = var.environment == "prod"
+  monitoring_enabled = var.monitoring || local.is_production
+}
+```
+
+</CodeBlockConfig>
+
+To reference the values from the `locals` block in your configuration, use `local.<NAME>` syntax. The `locals` block defines local values, and you reference those values on an object named `local`, which is singular.
+
+For example, in your configuration you can reference `local.resource_name` to name resources, and use `local.primary_public_subnet` with `local.monitoring_enabled` to configure a web server:
+
+<CodeBlockConfig highlight="4,5,8,14,17" hideClipboard>
+
+```hcl
+resource "aws_instance" "web" {
+  ami           = data.aws_ami.ubuntu.id
+  instance_type = var.instance_type
+  subnet_id     = local.primary_public_subnet
+  monitoring    = local.monitoring_enabled
+
+  tags = {
+    Name        = local.resource_name
+    Environment = var.environment
+  }
+}
+
+resource "aws_security_group" "web" {
+  name = "${local.resource_name}-sg"
+
+  tags = {
+    Name = "${local.resource_name}-security-group"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+You can access local values in the module where you define them, but not in other modules. However, you can pass a local value to a child module as an [argument](/terraform/language/modules/syntax#calling-a-child-module).
+
+Local values help avoid repetition and give a meaningful name to the value of an expression. However, they can make configuration harder to read because they obscure where values originate. Use local values in situations where you reuse a single value in many places to let you change a value in a single place, or when the value is the result of a complex expression.
```

Comment:
```suggestion
Local values help avoid repetition and give a meaningful name to the value of an expression. However, they can make configuration harder to read because they obscure where values originate. Use local values in situations where you either reuse a single value in many places to let you change a value in a single place or when the value is the result of a complex expression.
```